packages/twenty-front/src/modules/ui/input/components/SelectControl.tsx

```diff
@@ -45,18 +55,18 @@ export const SelectControl = ({
 
   return (
     <StyledControlContainer disabled={isDisabled}>
-      <StyledControlLabel>
-        {!!selectedOption?.Icon && (
+      {isDefined(selectedOption.Icon) ? (
+        <StyledIconContainer>
           <selectedOption.Icon
             color={
               isDisabled ? theme.font.color.light : theme.font.color.primary
             }
             size={theme.icon.size.md}
             stroke={theme.icon.stroke.sm}
           />
-        )}
-        {selectedOption?.label}
-      </StyledControlLabel>
+        </StyledIconContainer>
+      ) : null}
+      <StyledLabel>{selectedOption.label}</StyledLabel>
```

Comment:
would it make sense to use OverflowingTextWithTooltipe here?

Reply:
Done! It works great. However, I'm not sure how accessible using this component here is.